src/utils/wallet.ts

```diff
@@ -441,6 +366,142 @@ const wallet = {
     const xpriv = this.getXPrivKeyFromSeed(seed, methodOptions);
     return this.getMultiSigXPubFromXPriv(xpriv);
   },
+
+  generateAccessDataFromXpub(xpubkey: string, {multisig}: {multisig?: IMultisigData} = {}): IWalletAccessData {
+    let walletType: WalletType;
+    if (multisig === undefined) {
+      walletType = WalletType.P2PKH;
+    } else {
+      walletType = WalletType.MULTISIG;
+    }
+    // HDPublicKeys cannot derive on hardened paths, so the derivation must be done previously with the xprivkey.
+    // So we assume the user sent an xpub derived to the account level.
+    const accXPub = HDPublicKey(xpubkey);
+    const xpub = accXPub.derive(0);
```

Comment:
This was an issue I found when testing the many ways we can start the wallet.
The old method to start a wallet was not compatible with multisig since we require the public key on the account path.

It would be impossible to start a multisig wallet as a read-only, so I changed to only accept the account path xpub.
Do you think this would be troublesome?

Reply:
Made changes so that both account path and change path xpubkeys are accepted.
This makes the current behavior work while also adding support for multisig readonly wallets